Code/Mantid/Framework/API/src/Projection.cpp

```diff
@@ -38,6 +40,64 @@ Projection::Projection(const V3D &u, const V3D &v, const V3D &w) {
   }
 }
 
+Projection::Projection(ITableWorkspace_const_sptr ws) {
+  if (!ws)
+    throw std::runtime_error(
+        "Null ITableWorkspace given to Projection constructor");
+
+  if (ws->columnCount() != 4)
+    throw std::runtime_error(
+        "4 columns must be provided to create a projection");
+
+  const size_t numRows = ws->rowCount();
+  if (numRows != 3)
+    throw std::runtime_error("3 rows must be provided to create a projection");
+
+  for (size_t i = 0; i < numRows; i++) {
+    const std::string name = ws->getColumn("name")->cell<std::string>(i);
+    const std::string valueStr = ws->getColumn("value")->cell<std::string>(i);
+    const double offset = ws->getColumn("offset")->cell<double>(i);
+    const std::string unitStr = ws->getColumn("type")->cell<std::string>(i);
+
+    //Check the name
+    size_t index;
+    if (name == "u") {
+      index = 0;
+    } else if (name == "v") {
+      index = 1;
+    } else if (name == "w") {
+      index = 2;
+    } else {
+      throw std::runtime_error("Invalid dimension name: " + name);
+    }
+
+    // Check the values
+    std::vector<std::string> valueStrVec;
+    boost::split(valueStrVec, valueStr, boost::is_any_of(","));
+    std::vector<double> valueDblVec;
+    for (auto it = valueStrVec.begin(); it != valueStrVec.end(); ++it)
+      valueDblVec.push_back(boost::lexical_cast<double>(*it));
+    if (valueDblVec.size() != 3) {
+      throw std::runtime_error("Dimension " + name + " must contain 3 values");
+    }
```

Comment:
The values are placed into a V3D further down, on line 95. The reason for using a vector here is so we can leverage `boost::split`. I think it's clearer to split and cast up here as a single step first before sanity checking the data and constructing the V3D.